lib/full-site-editing/block-templates.php

```diff
@@ -137,6 +137,35 @@ function _gutenberg_add_template_part_area_info( $template_info ) {
 	return $template_info;
 }
 
+/**
+ * Returns an array containing the references of
+ * the passed blocks and their inner blocks.
+ *
+ * @param array $blocks array of blocks.
+ *
+ * @return array block references to the passed blocks and their inner blocks.
+ */
+function _get_blocks_recursive( &$blocks ) {
+	$all_blocks = array();
+	$queue      = array();
+	foreach ( $blocks as &$block ) {
+		$queue[] = &$block;
+	}
+
+	for ( $i = 0; $i < count( $queue ); $i++ ) {
```

Comment:
I found growing the queue size in the loop a bit weird, but couldn't think of a better way to deal with these that would preserve the references.

Reply:
I found it a little odd at first too, but after a minute it sits pretty well imo.  

> couldn't think of a better way to deal with these that would preserve the references.

Perhaps the recursive function approach would still work here since we are assigning by reference?  I don't think its necessary though.  This reads just as well as that would.

Reply:
I prefer iteration over recursion. That's why the growing queue array.

Reply:
> I prefer iteration over recursion. 

I tend to agree with you on that. 😁